Description: vidir: create missing target directories
 The ability to use vidir to move files to a non-existing directory
 (hierarchy) is enabled by this patch.  Missing directories are created
 on-the-fly (with optional verbose output).
Author: Nicolas Schier <nicolas@hjem.rpa.no>
Bug-Debian: https://bugs.debian.org/728688
Forwarded: no
Last-Update: <2015-04-17>

--- moreutils-0.55.orig/vidir
+++ moreutils-0.55/vidir
@@ -76,6 +76,8 @@ Licensed under the GNU GPL.
 
 =cut
 
+use File::Basename;
+use File::Path qw(make_path);
 use File::Spec;
 use File::Temp;
 use Getopt::Long;
@@ -152,7 +154,8 @@ while (<IN>) {
 		elsif ($name ne $item{$num}) {
 			next unless length $name;
 			my $src=$item{$num};
-			
+			my $dir=dirname($name);
+
 			if (! (-e $src || -l $src) ) {
 				print STDERR "$0: $src does not exist\n";
 				delete $item{$num};
@@ -181,7 +184,13 @@ while (<IN>) {
 				}
 			}
 
-			if (! rename($src, $name)) {
+			if ((! -d $dir) && (! make_path($dir, {
+							verbose => $verbose,
+						}))) {
+				print STDERR "$0: failed to create directory tree $dir: $!\n";
+				$error=1;
+			}
+			elsif (! rename($src, $name)) {
 				print STDERR "$0: failed to rename $src to $name: $!\n";
 				$error=1;
 			}
